### PR TITLE
Cutting the Burdened Cheese Part 1: Removes free burden points from removing unnecessary or cosmetic organs.

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Burdened organs now only count for the main organs only. Cyberimplants don't count, alien organs don't count, etc. This is because someone could get a bunch of extra organs, and then take the burdened sect and remove them for free burden points.

Useless organs for species also do not count (stomachs given to species that do not hunger, for instance)

Removes the error on burden points clamping on negative values. This can happen when someone previously disabled takes burden sect.

## Why It's Good For The Game

Cheese is being used to bypass the unique gameplay goals burdened sect provides. Will throw another pr when fikou finishes bat mutants ;)

## Changelog

:cl:
fix: Removed some cheese strategies from burdened sect.
/:cl:

### CODE REVIEW

File: code/modules/religion/burdened/burdened_trauma.dm [hunk 1]
```diff
@@ -143,13 +143,33 @@
 	if(special) //aheals
 		return
 
-	if(istype(old_organ, /obj/item/organ/internal/eyes))
+	/// only organs that are slotted in these count. because there's a lot of useless organs to cheese with.
+	var/list/slots_youd_actually_miss = list(
```
Comment (Autisem) [inline code/modules/religion/burdened/burdened_trauma.dm:147]: Maby renaming it to "critical_slots"? idk the names a bit odd

**Comment by Pickle-Coding** [comment]
You put your answer for why it's good for the game inside the changelog instead of Why It's Good For The Game section.

**Comment by tralezab** [comment]
> You put your answer for why it's good for the game inside the Changelog section instead of Why It's Good For The Game section.

thanks for catching that

**Comment by tralezab** [comment]
> What if someone implants an alien tongue, they can be found in maints.

The only exploit would be putting in the tongue on a species that doesn't have a tongue by default, and then taking it out with burdened. We don't have a species that spawns without a tongue.

File: code/modules/religion/burdened/burdened_trauma.dm [hunk 2]
```diff
@@ -143,13 +143,33 @@
...
+		ORGAN_SLOT_EYES,
+		ORGAN_SLOT_EARS,
+		ORGAN_SLOT_TONGUE,
+		ORGAN_SLOT_LIVER,
+		ORGAN_SLOT_STOMACH,
+	)
+	var/list/bad_traits = list(
+		TRAIT_NOHUNGER = ORGAN_SLOT_STOMACH,
+		TRAIT_NOBREATH = ORGAN_SLOT_LUNGS,
+		TRAIT_NOMETABOLISM = ORGAN_SLOT_LIVER,
```
Comment (ATH1909) [inline code/modules/religion/burdened/burdened_trauma.dm:160]: what happens if you get these traits AFTER you've already removed these organs? plasmamen, for example, get their NOHUNGER trait from their lungs- would removing their lungs before their stomach give more points than the other way around?

**Comment by ATH1909** [comment]
> The only exploit would be putting in the tongue on a species that doesn't have a tongue by default, and then taking it out with burdened. We don't have a species that spawns without a tongue.

cardboard golems, IIRC

File: code/modules/religion/burdened/burdened_trauma.dm (hunk 2 above)
Comment (tralezab) [reply code/modules/religion/burdened/burdened_trauma.dm:160]: If you have the traits, the organ slots simply don't work with the system. In theory, there could be an exploit if you were able to repeatedly remove the organ, add the trait, remove the trait, and remove the organ again, but... do we even have a way to gain and remove these traits controllably and repeatably?

**Comment by tralezab** [comment]
> > The only exploit would be putting in the tongue on a species that doesn't have a tongue by default, and then taking it out with burdened. We don't have a species that spawns without a tongue.
> 
> cardboard golems, IIRC

Wow damn guess we do

**Comment by Overuseddart** [comment]
What about species that dont need a heart or lungs? Or characters that start with posi limbs

File: code/modules/religion/burdened/burdened_trauma.dm (hunk 2 above)
Comment (ATH1909) [reply code/modules/religion/burdened/burdened_trauma.dm:157]: > If you have the traits, the organ slots simply don't work with the system. In theory, there could be an exploit if you were able to repeatedly remove the organ, add the trait, remove the trait, and remove the organ again, but... do we even have a way to gain and remove these traits controllably in a repeatable situation?

wouldn't removing/readding plasmaman lungs allow you to manipulate the nohunger trait independently of your stomach?

File: code/__DEFINES/traits.dm [hunk 3]
```diff
@@ -214,7 +214,14 @@ Remember to update _globalvars/traits.dm if you're adding/removing/renaming trai
...
 #define TRAIT_NOSELFIGNITION_HEAD_ONLY "no_selfignition_head_only"
 #define TRAIT_NOGUNS "no_guns"
+///This carbon doesn't get hungry, has no stomach or appendix
 #define TRAIT_NOHUNGER "no_hunger"
+///This carbon doesn't bleed, has no heart
+#define TRAIT_NOBLOOD "noblood"
+///This carbon doesn't have an appendix. Which is funny, because that doesn't change a whole lot out of dodging appendicitis
+#define TRAIT_NOAPPENDIX "no_appendix"
+///This carbon doesn't have a tongue.
+#define TRAIT_NOTONGUE "no_tongue"
```
Comment (Fikou) [inline code/__DEFINES/traits.dm:224]: these shouldnt be mob traits, bring em back to species traits, even if theyre stupid

**Comment by tralezab** [comment]
picking this  up again

File: code/modules/antagonists/cult/blood_magic.dm [hunk 4]
```diff
@@ -676,7 +676,7 @@
 	if(proximity)
 		if(ishuman(target))
 			var/mob/living/carbon/human/H = target
-			if(NOBLOOD in H.dna.species.species_traits)
+			if(HAS_TRAIT(H, TRAIT_NOBLOOD))
 				to_chat(user,span_warning("Blood rites do not work on species with no blood!"))
```
Comment (Fikou) [inline code/modules/antagonists/cult/blood_magic.dm:680]:
```suggestion
				to_chat(user,span_warning("Blood rites do not work on people with no blood!"))
```

**Pull request merged**